rayon-core/src/thread_pool/mod.rs

```diff
@@ -161,8 +212,43 @@ impl ThreadPool {
         unsafe { spawn::spawn_in(op, &self.registry) }
     }
 
-    /// Spawns an asynchronous future in this thread-pool. See
-    /// `spawn_future()` for more details.
+    /// Spawns an asynchronous future in the thread pool. `spawn_future()` will inject 
+    /// jobs into the threadpool that are not tied to your current stack frame. This means 
+    /// `ThreadPool`'s `spawn` methods are not scoped. As a result, it cannot access data
+    /// owned by the stack.
+    ///
+    /// `spawn_future()` returns a `RayonFuture<F::Item, F::Error>`, allowing you to chain
+    /// multiple jobs togther.
+    ///
+    /// ## Using `spawn_future()`
+    ///
+    /// ```rust
+    ///    # extern crate rayon_core as rayon;
+    ///    extern crate futures;
+    ///    
+    ///    use futures::{future, Future};
+    ///    use std::sync::{Arc, Mutex};
+    ///    # fn main() {
+    ///    let data = Arc::new(Mutex::new(String::from("Hello,")));
+    ///    let pool = rayon::ThreadPool::new(rayon::Configuration::new().num_threads(8)).unwrap();
+    ///
+    ///    let a = pool.spawn_future(future::lazy({
+    ///                                               let data = data.clone();
+    ///                                               move || Ok::<_, ()>(data)
+    ///                                           }));
+    ///    let b = pool.spawn_future(a.map(|data| {
```

Comment:
This is a bit of a funny example -- in particular, futures are best used when they **produce a result that you want to consume**, but this example works via side-effects. I would recommend that code like this be written with plain spawns, rather than futures (in particular, the cancellation semantics of futures could lead to surprising interactions).

Maybe rewrite this to the following:

```rust
let a = pool.spawn_future(future::lazy(move || Ok::<_, ()>(format!("Hello, world!")));
let b = pool.spawn_future(a.map(|mut data| { data.push_str("world"); data }));
let result = b.wait().unwrap(); // `Err` is impossible, so use `unwrap()` here
println!("{:?}", result); // prints: "Hello, world!"
```